src/operations/execute_operation.ts

```diff
@@ -251,7 +254,10 @@ async function retryOperation<
   }
 
   // select a new server, and attempt to retry the operation
-  const server = await topology.selectServerAsync(selector, { session });
+  const server = await topology.selectServerAsync(selector, {
+    session: session,
```

Comment:
```suggestion
    session,
```

Reply:
Didn't catch that, should be changed now.